platform/source/mbed_power_mgmt.c

```diff
@@ -239,6 +243,9 @@ void sleep_manager_sleep_auto(void)
 #endif
         hal_deepsleep();
     } else {
+#if defined(MBED_SLEEP_STAT_ENABLED)
```

Comment:
you get stats at the top of this function, do we need to print  this again when only sleeping?